feat/src/main/java/com/example/feat/course_detail/CourseDetailScreen.kt

```diff
@@ -0,0 +1,267 @@
+package com.example.feat.course_detail
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Info
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.example.core.components.Skeleton
+import com.example.core.extensions.toEEEEDDMMYYYY
+import com.example.core.lib.constants.DesignSystem
+import com.example.domain.entity.TopicEntity
+import com.example.domain.mapper.toExperienceText
+import com.example.feat.R
+import com.example.feat.course_detail.components.CourseDetailToolbar
+import com.example.feat.course_detail.controller.CourseDetailUiEvent
+import com.example.feat.course_detail.controller.CourseDetailViewModel
+
+@Composable
+fun CourseDetailScreen(
+    courseId: String,
+    courseDetailViewModel: CourseDetailViewModel = hiltViewModel()
+) {
+    val lazyScrollState = rememberLazyListState()
+    val uiState = courseDetailViewModel.uiState.collectAsState()
+    LaunchedEffect(Unit) {
+        courseDetailViewModel.onEvent(CourseDetailUiEvent.FetchCourseDetail(courseId))
+    }
```

Comment:
The `LaunchedEffect` with `Unit` key will run only once when the `CourseDetailScreen` composable enters the composition. If you need to refetch course details based on `courseId`, consider using `LaunchedEffect(courseId)` instead.

```diff
-LaunchedEffect(Unit) {
+LaunchedEffect(courseId) {
    courseDetailViewModel.onEvent(CourseDetailUiEvent.FetchCourseDetail(courseId))
```